app/proxy.go

```diff
@@ -117,6 +158,11 @@ func setKeepalive(conn *net.TCPConn) error {
 			if err != nil {
 				return
 			}
+
+			err = syscall.SetsockoptInt(fd, syscall.IPPROTO_TCP, unix.TCP_USER_TIMEOUT, int(30*time.Second/time.Millisecond))
```

Comment:
Adding a comment here detailing some of the reason behind its presence (from the comment in LXD's `SetTCPTimeouts`) would likely be beneficial for future readers.